src/battle_anim_effects_1.c

```diff
@@ -6642,6 +6642,79 @@ static void ReloadBattlerSprites(u32 battler, struct Pokemon *party)
     }
 }
 
+static void TrySwapSkyDropTargets(u32 battlerAtk, u32 battlerPartner)
+{
+    u32 i, temp;
+    
+    // battlerAtk is using Ally Switch
+    // check if our partner is the target of sky drop
+    // If so, change that index to battlerAtk
+    for (i = 0; i < gBattlersCount; i++) {
+        if (gBattleStruct->skyDropTargets[i] == battlerPartner) {
+            gBattleStruct->skyDropTargets[i] = battlerAtk;
+            break;
+        }
+    }
+    
+    // Then swap our own sky drop targets with the partner in case our partner is mid-skydrop
+    SWAP(gBattleStruct->skyDropTargets[battlerAtk], gBattleStruct->skyDropTargets[battlerPartner], temp);
+}
+
+#define TRY_SIDE_TIMER_BATTLER_ID_SWAP(battlerAtk, battlerPartner, side, field)    \
+    if (gSideTimers[side].field == battlerAtk)                      \
+        gSideTimers[side].field = battlerPartner;                   \
+    else if (gSideTimers[side].field == battlerPartner)             \
+        gSideTimers[side].field = battlerAtk;
+
+static void TrySwapStickyWebBattlerId(u32 battlerAtk, u32 battlerPartner)
+{
+    u32 atkSide = GetBattlerSide(battlerAtk);
+    u32 oppSide = GetBattlerSide(BATTLE_OPPOSITE(battlerAtk));
+    
+    // not all of these are needed to be swapped, but are done so to be robust to anything in the future that might care about them
+    TRY_SIDE_TIMER_BATTLER_ID_SWAP(battlerAtk, battlerPartner, atkSide, reflectBattlerId);
+    TRY_SIDE_TIMER_BATTLER_ID_SWAP(battlerAtk, battlerPartner, atkSide, lightscreenBattlerId);
+    TRY_SIDE_TIMER_BATTLER_ID_SWAP(battlerAtk, battlerPartner, atkSide, mistBattlerId);
+    TRY_SIDE_TIMER_BATTLER_ID_SWAP(battlerAtk, battlerPartner, atkSide, safeguardBattlerId);
+    TRY_SIDE_TIMER_BATTLER_ID_SWAP(battlerAtk, battlerPartner, atkSide, auroraVeilBattlerId);
+    TRY_SIDE_TIMER_BATTLER_ID_SWAP(battlerAtk, battlerPartner, atkSide, tailwindBattlerId);
+    TRY_SIDE_TIMER_BATTLER_ID_SWAP(battlerAtk, battlerPartner, atkSide, luckyChantBattlerId);
+    
+    // if we've set sticky web on the opposing side, need to swap stickyWebBattlerId for mirror armor
+    TRY_SIDE_TIMER_BATTLER_ID_SWAP(battlerAtk, battlerPartner, oppSide, stickyWebBattlerId);
+}
+#undef TRY_SIDE_TIMER_BATTLER_ID_SWAP
+
+static void TrySwapWishBattlerIds(u32 battlerAtk, u32 battlerPartner)
+{
+    u32 i, temp;
+    u32 oppSide = GetBattlerSide(BATTLE_OPPOSITE(battlerAtk));
+    
+    // if used future sight on opposing side, properly track who used it
```

Comment:
is this according to vanilla games?

Reply:
Hm good question I guess I just assumed but might need testing

Reply:
[Future Sight targets aren't switched](https://replay.pokemonshowdown.com/gen7doublesou-2263137865). Tested on showdown but very likely it is the same in vanilla games

Reply:
In the clip Gothitelle was originally targeted? Or the opposing Gardevoir and Gothitelle took it due to their ally switch? The code here only intends to update the attacker battler for damage calculation, so I think that is consistent? Maybe a test would help, but I might need assistance with it. I'm less proficient at testing damage calc scenarios

Reply:
oh I see. Ye, that is probably correct?

Reply:
Yep, this is correct unless it differs in vanilla games. [Showdown replay](https://replay.pokemonshowdown.com/gen7doublesou-2263334835) User and target were both Gardevoir